Update Pagination in Blog Index Page

oz456 opened the pull request:
**Description:**
---
This PR updates the pagination system on the blog index page, as referenced in #349. It introduces pagination logic that aligns with the conventions outlined in the GSC docs.

 **Pagination Sequence Logic:**
---
  - If there are 7 or fewer pages in total, all pages are displayed without ellipses (`...`).
  - For more than 7 pages:
    - Displays up to 7 pages at a time, with ellipses (`...`) when necessary.
    - Displays 6 pages when the current page is near either end of the page range.
    - Always displays at least 2 page numbers to the left and right of the current page, if they exist.

 **Screenshots:**
---
<details><summary>Pagination Sequence on different pages</summary>
<p>

For more than 7 pages in total:
![Screenshot from 2025-03-23 02-12-44 (1)](https://github.com/user-attachments/assets/1a2bf34d-a39b-4aeb-9fea-d3b9667e35fe)
![Screenshot from 2025-03-23 02-12-38](https://github.com/user-attachments/assets/aee4360e-4e0e-450a-a172-063920d1e6f2)
![Screenshot from 2025-03-23 02-11-54](https://github.com/user-attachments/assets/5a093103-38ae-42bb-9b52-1ede958b9cba)
![Screenshot from 2025-03-23 02-11-43](https://github.com/user-attachments/assets/90c0f057-3c8e-49ed-a5dc-cf075e750a85)
![Screenshot from 2025-03-23 02-11-21](https://github.com/user-attachments/assets/148addab-3270-455a-8059-f96dc88e21cf)
For 7 or fewer total pages:
![Screenshot from 2025-03-23 02-10-01](https://github.com/user-attachments/assets/907481cc-6a30-40bc-93ba-0a73466535fe)
![Screenshot from 2025-03-23 09-03-04](https://github.com/user-attachments/assets/8a59f484-ab31-4552-8adb-1fbb91372c62)

</p>
</details> 

<details><summary>Pagination on Small devices</summary>
<p>

![Screenshot from 2025-03-23 02-16-47 (2)](https://github.com/user-attachments/assets/7f2fb716-d680-4bc1-a5ee-aaa64e51b1bf)

</p>
</details>

zerolab commented:
It may be worth checking out https://github.com/wagtail/wagtail/pull/12607 which addresses a similar requirement, but for the Wagtail admin

vossisboss commented:
@oz456 Thank you SO much for this PR! We appreciate you taking the time to make this contribution. I'll try to take a lot at this before the end of the week to test your work and give you some feedback.

Thanks again!

vossisboss commented:
All right @oz456, I took a look at this and it seems like your styles need some work because this is how the pagination look for me both in the Firefox and Chrome browsers.

![Screenshot 2025-03-26 at 5 56 53 PM](https://github.com/user-attachments/assets/3f844264-258d-490c-964c-466595da96ba)

![Screenshot 2025-03-26 at 5 57 38 PM](https://github.com/user-attachments/assets/0ff384c7-2788-4a4d-8523-5eb220ecba4f)

The feature was functional, which was great. But if you could working on fixing the styles and submit again, I'd really appreciate it.

oz456 commented:
@vossisboss, sorry to hear that it's not working as expected. However, I am able to view it properly on my end.

![Screenshot from 2025-03-27 09-51-36(1)](https://github.com/user-attachments/assets/d4662ed2-a3e1-4d9f-aed1-401fdd40df41)

I'm not sure why this is happening. but just in case, did you recompile the styles?

vossisboss commented:
Excellent suggestion @oz456 because my `npm` was not auto-refreshing like it was supposed to. I'm now seeing what you are. Looks like you've done a great job with preserving the animations and the keyboard navigation works well.  At first glance, it seems you've also made some good choices with ARIA labels as well.

Could you restore the "Previous" and "Next" labels to the buttons? I feel like those labels are helpful, especially for users who use assistive technology.

Thanks again for this submission. I appreciate the time you're taking with this. If you haven't joined the #wagtail-org channel in our Slack community yet, that is a good way to get in touch with other community members who contribute to the website.

oz456 commented:
Thanks @vossisboss for reviewing this PR.

The reason I initially removed the labels from the buttons was that they were causing issues in certain viewports, especially at specific edge cases. Removing the labels seemed to be the most efficient solution at that time.

<details><summary>Previous Issue </summary>
<p>

![Screenshot from 2025-03-28 19-40-54](https://github.com/user-attachments/assets/6d2c103a-0716-4c25-969b-277133184573)

</p>
</details> 

However, I’ve added back the labels and addressed the issue by aligning the buttons in a column layout (similar to the mobile view) until the point where the layout breaks. I'm not entirely sure if the styles are optimal yet, so let me know if they look good or if there's anything that can be improved.

Stormheg reviewed (approved):
Thank you! Sorry to keep your PR open so long, it is a welcome improvement and we should have reviewed & merged it sooner 👍
